src/IceModel.cpp

```diff
@@ -0,0 +1,168 @@
+#include <radiopropa/IceModel.h>
+
+namespace radiopropa {
+
+ExponentialIndex::ExponentialIndex(double n_ice, double delta_n, double z_0, double z_shift): 
+	_n_ice(n_ice), 
+	_delta_n(delta_n), 
+	_z_shift(z_shift), 
+	_z_0(z_0)
+{}
+ExponentialIndex::~ExponentialIndex()
+{}
+double ExponentialIndex::getValue(const Vector3d &position) const
+{
+	return _n_ice  - _delta_n  * exp((position.z-_z_shift) / _z_0);
+}
+double ExponentialIndex::getAverageValue(const Vector3d &position1, const Vector3d &position2) const
+{
+	return _n_ice - _delta_n * _z_0 / (position2.z - position1.z) * (exp((position2.z-_z_shift) / _z_0) - exp((position1.z-_z_shift) / _z_0));
+}
+Vector3d ExponentialIndex::getGradient(const Vector3d &position) const
+{
+	Vector3d v(0,0,0);
+	v.z = - _delta_n / _z_0 * exp((position.z-_z_shift)/ _z_0);
+	return v;
+}
+
+
+IceModel_Simple::IceModel_Simple(double n_ice, double delta_n, double z_0, double z_shift, double z_surface): 
+	_z_surface(z_surface),
+	_ice(n_ice, delta_n, z_0, z_shift)
+{}
+IceModel_Simple::~IceModel_Simple()
+{}
+double IceModel_Simple::getValue(const Vector3d &position) const
+{
+	if (position.z <= _z_surface) {
+		return _ice.getValue(position);
+	} else {
+		return 1.;
+	}
+}
+double IceModel_Simple::getAverageValue(const Vector3d &position1, const Vector3d &position2) const
+{
```

Comment:
What is the idea of getting the average for 2 positions of an analytic function? For what is this used?

Reply:
It was implemented in medium.py of NuRadioMC, so I stream lined it